iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/MoveFeederBay.java

```diff
@@ -0,0 +1,132 @@
+    private final String connectableId;
+    private final String targetBusOrBusbarSectionId;
+    private final String targetVoltageLevelId;
+    private final Terminal terminal;
+
+    /**
+     * Constructor
+     * @param connectableId non-null id of the connectable whose feeder bay will be moved (busbar sections are not accepted)
+     * @param terminal non-null id of the terminal
+     * @param targetBusbarSectionId non-null id of the target busbar section
+     * @param targetVoltageLevelId non-null id of the target voltage Level
+     */
+    public MoveFeederBay(String connectableId, String targetBusbarSectionId, String targetVoltageLevelId, Terminal terminal) {
+        this.connectableId = Objects.requireNonNull(connectableId);
+        this.targetBusOrBusbarSectionId = Objects.requireNonNull(targetBusbarSectionId);
+        this.targetVoltageLevelId = Objects.requireNonNull(targetVoltageLevelId);
+        this.terminal = Objects.requireNonNull(terminal);
+    }
+
+    @Override
+    public void apply(Network network, NamingStrategy namingStrategy, boolean throwException, ComputationManager computationManager, ReportNode reportNode) {
+        Connectable<?> connectable = network.getConnectable(connectableId);
+        if (!checkConnectable(throwException, reportNode, connectable)) {
+            return;
+        }
+        VoltageLevel voltageLevel = network.getVoltageLevel(targetVoltageLevelId);
+        if (voltageLevel.getTopologyKind() == TopologyKind.NODE_BREAKER) {
+            cleanNodeBreakerTopology(network, connectableId, reportNode);
+            createTopology(getSideFromTerminal(terminal, connectable), targetBusOrBusbarSectionId, network, voltageLevel, connectable, namingStrategy, reportNode);
+            Map<VoltageLevel, Integer> firstAvailableNodes = new HashMap<>();
+            int connectableNode = firstAvailableNodes.compute(voltageLevel, this::getNextAvailableNode);
+            terminal.getNodeBreakerView().moveConnectable(connectableNode, voltageLevel.getId());
+        } else {
+            terminal.getBusBreakerView().moveConnectable(targetBusOrBusbarSectionId, terminal.isConnected());
+        }
+    }
+
+    private int getSideFromTerminal(Terminal terminal, Connectable<?> connectable) {
+        if (connectable instanceof Injection<?>) {
+            return 0;
+        } else if (connectable instanceof Branch<?>) {
+            return terminal.getSide().getNum();
+        }
+        throw new IllegalStateException("Unexpected connectable: " + connectable);
+    }
+
+    private int getNextAvailableNode(VoltageLevel vl, Integer node) {
+        return node == null ? vl.getNodeBreakerView().getMaximumNodeIndex() + 1 : node + 1;
+    }
+
+    @Override
+    public String getName() {
+        return "MoveFeederBay";
+    }
+
+    @Override
+    public NetworkModificationImpact hasImpactOnNetwork(Network network) {
+        NetworkModificationImpact impact = DEFAULT_IMPACT;
```

Comment:
To follow the same principle as the other `NetworkModification`s, you should update the `AbstractNetworkModification#impact` attribute. So you don't have to define a local variable:
```suggestion
        impact = DEFAULT_IMPACT;
```